packages/mui-material/src/Radio/Radio.js

```diff
@@ -53,7 +53,7 @@ const RadioRoot = styled(SwitchBase, {
       {
         props: { color: 'default', disableRipple: false },
         style: {
-          '&:hover': {
+          [`&:not(.${radioClasses.disabled})&:hover`]: {
```

Comment:
This increases the specificity compared to v5:

https://github.com/mui/material-ui/blob/6dcf4d6f6b2b643de26537233e29e86b240c5a72/packages/mui-material/src/Radio/Radio.js#L46